openedx/core/djangoapps/content_tagging/rest_api/v1/views.py

```diff
@@ -130,8 +142,91 @@ def orgs(self, request, **_kwargs) -> Response:
 class ObjectTagOrgView(ObjectTagView):
     """
     View to create and retrieve ObjectTags for a provided Object ID (object_id).
-    This view extends the ObjectTagView to add Organization filters for the results.
 
     Refer to ObjectTagView docstring for usage details.
     """
     filter_backends = [ObjectTagTaxonomyOrgFilterBackend]
+
+
+class ObjectTagExportView(APIView):
+    """"
+    View to export a CSV with all children and tags for a given object_id.
+    """
+    def get(self, request: Request, **kwargs) -> StreamingHttpResponse:
+        """
+        Export a CSV with all children and tags for a given object_id.
+        """
+
+        class Echo(object):
+            """
+            Class that implements just the write method of the file-like interface,
+            used for the streaming response.
+            """
+            def write(self, value):
+                return value
+
+        def _generate_csv_rows(
+            tagged_content: TaggedContent,
+            taxonomies: TaxonomyDict,
+            pseudo_buffer: Echo,
+        ) -> Iterator[str]:
+            """
+            Receives the blocks, tags and taxonomies and returns a CSV string
+            """
+
+            header = {"name": "Name", "type": "Type", "id": "ID"}
+
+            # Prepare the header for the taxonomies
+            for taxonomy_id, taxonomy in taxonomies.items():
+                # ToDo: change to taxonomy.external_id after the external_id is implemented
+                header[f"taxonomy_{taxonomy_id}"] = taxonomy.name
```

Comment:
note: this is blocked on https://github.com/openedx/edx-platform/pull/34143

Reply:
Done: c82e9cb2360c851b16bab219d671d04ad17770c8